File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_13_chain_of_responsibilities/java/_02_after/LoggingRequestHandler.java
package com.example._03_behavioral_patterns._13_chain_of_responsibilities.java._02_after;

import com.example._03_behavioral_patterns._13_chain_of_responsibilities.java._01_before.Request;

public class LoggingRequestHandler extends RequestHandler {

    public LoggingRequestHandler(RequestHandler nextHandler) {
        super(nextHandler);
    }

    @Override
    public void handle(Request request) {
        System.out.println("로깅");
        super.handle(request);
    }
}

File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_13_chain_of_responsibilities/kt/_02_after/LoggingRequestHandler.kt
package com.example._03_behavioral_patterns._13_chain_of_responsibilities.kt._02_after

import com.example._03_behavioral_patterns._13_chain_of_responsibilities.kt._01_before.Request

class LoggingRequestHandler(nextHandler: RequestHandler?) : RequestHandler(nextHandler) {

    override fun handle(request: Request) {
        println("로깅")
        super.handle(request)
    }
}
